mono/metadata/reflection.c

```diff
@@ -10397,22 +10503,26 @@ ctor_builder_to_signature (MonoImage *image, MonoReflectionCtorBuilder *ctor) {
  */
 static MonoMethodSignature*
 method_builder_to_signature (MonoImage *image, MonoReflectionMethodBuilder *method) {
+	MonoError error;
 	MonoMethodSignature *sig;
 
 	sig = parameters_to_signature (image, method->parameters);
 	sig->hasthis = method->attrs & METHOD_ATTRIBUTE_STATIC? 0: 1;
-	sig->ret = method->rtype? mono_reflection_type_get_handle ((MonoReflectionType*)method->rtype): &mono_defaults.void_class->byval_arg;
+	sig->ret = method->rtype? mono_reflection_type_get_handle ((MonoReflectionType*)method->rtype, &error): &mono_defaults.void_class->byval_arg;
```

Comment:
This is bad. The error might be uninitialized if method->rtype is NULL.

Similar issues on other places.